core/server/master/src/test/java/alluxio/master/file/FileSystemMasterTest.java

```diff
@@ -918,6 +918,41 @@ public void listStatus() throws Exception {
     }
   }
 
+  @Test
+  public void listStatusRecursive() throws Exception {
+    final int files = 10;
+    List<FileInfo> infos;
+    List<String> filenames;
+
+    // Test files in root directory.
+    for (int i = 0; i < files; i++) {
+      createFileWithSingleBlock(ROOT_URI.join("file" + String.format("%05d", i)));
+    }
+    // Test files in nested directory.
```

Comment:
does this create multi-level nested directories?

Reply:
yes